src/v/archival/service.cc

```diff
@@ -193,12 +193,18 @@ ss::future<> scheduler_service_impl::start() {
 }
 
 ss::future<> scheduler_service_impl::stop() {
+    if (_stopped) {
+        vlog(_rtclog.info, "Scheduler service is already stopped");
+        co_return;
+    }
+
     vlog(_rtclog.info, "Scheduler service stop");
     _timer.cancel();
     co_await _gate.close();
     for (auto& it : _archivers) {
         co_await it.second->stop();
     }
+    _stopped = true;
```

Comment:
if there is a risk that this is called twice, is there a risk that it's called before `_stopped` is set to true, since there are some scheduling points before it and after it is checked at the beginning?

Reply:
This is called twice in unit tests where we block for stop to complete, but I agree it makes more sense to set `_stopped` to true earlier, so I have moved it up.